### PR TITLE
{ WIP } Update Context page and rename it Additional Data

### PR DESCRIPTION
**Pull request by MimiDumpling**
WORK IN PROGRESS

The Context page needed a revamp with more focus on the additional data a user can apply to Sentry rather than the overly vague concept of context. The content of this page now serves as general guidance of additional data.

To Do:

- [x] fix all links referring to Context and change to Additional Data
- [x] fix redirects for anything referring to Context
- [x] update links within the page
- [x] compress and add screenshots
- [x] update snapshot

### CODE REVIEW

**Comment by untitaker** [comment]
Please keep in mind that Additional Data is already a used term in the UI specifically for 'extra' in the event. I fear using this word might confuse people just as using context does

**Comment by adhiraj** [comment]
I would rather change it in the app as well to say custom context or custom data. Any suggestion/preference?

**Comment by untitaker** [comment]
If you want to start such a mass-rename, I would prefer if we renamed all sections in the UI and in the documentation to say "extra" for "extra" and "context" for "context". It is already in conflict with SDKs which have `setExtra` as methods whose data ends up in "Additional Data", and `setContext` for data that now also ends up in "Custom Data" as you called it.

Maybe I'm not seing the full picture here but this seems to pile onto a situation that is already bad.

**Comment by bruno-garcia** [comment]
Please take into consideration the name in protocol iself, `Contexts`, which is typed in to many of our SDKs:

SDK Docs:
https://docs.sentry.io/development/sdk-dev/event-payloads/contexts/

Examples:
https://github.com/getsentry/sentry-dotnet/blob/f6e40b52d2ea82fc513929a7cf8ba6e046ea714f/src/Sentry.Protocol/Context/Contexts.cs#L13-L41

https://github.com/getsentry/sentry-android/blob/3141cb8ac16c52788832f21d2486945af2d5daa6/sentry-core/src/main/java/io/sentry/core/protocol/Contexts.java#L5-L59

**Comment by adhiraj** [comment]
@untitaker: 1) @dcramer mentioned that setExtra should be deprecated, and everyone should be using setContext going forward, 2) what are the other places in the UI (besides issue page additional data section) that need to be updated?

@bruno-garcia yep. to clarify a bit more, we're not changing context -> additional data, we're using the word "additional data" because that's how users think of it, and making it clear that context is one way to set additional data (another way being tags).

**Comment by bruno-garcia** [comment]
Happy with that. Just thought one might want to be aware for context and also might be worth changing the protocol docs to use in addition to `Contexts`, the new wording to help finding things on the docs search.

**Comment by bruno-garcia** [comment]
Happy with that. Just thought one might want to be aware for context and also might be worth changing the protocol docs to use in addition to `Contexts`, the new wording to help finding things on the docs search.

**Comment by untitaker** [comment]
>1) @dcramer mentioned that setExtra should be deprecated, and everyone should be using setContext going forward

Still feel like it's way too early to start reusing terms when practically nobody migrated yet.

>what are the other places in the UI (besides issue page additional data section) that need to be updated?

Likely the events page reachable from discover

FWIW I think it would be best if we have a comprehensive proposal on which changes are planned to the UI and which changes are planned to the wording on the docs, and then we review and apply all changes at once. I fear we will get stuck with this migration halfway through.

File: src/collections/_documentation/enriching-error-data/context.md [hunk 1]
```diff
@@ -8,17 +8,37 @@ example_extra_key: character_name
...
 
-Sending users to Sentry will unlock a number of features, primarily the ability to drill down into the number of users affecting an issue, as well to get a broader sense about the quality of the application.
+[Custom data](#custom-data) is arbitrary structured or unstructured extra data you can attach to your event.
 
-Capturing the user is fairly straight forward:
+## Tags & Context
+
+Whether predefined or custom, additional data can take two forms in Sentry: tags and context.
+
+Tags are key/value string pairs that are indexed and searchable. They power UI features like filters and tag-distribution heat maps.
```
Comment (markstory) [inline src/collections/_documentation/enriching-error-data/context.md:21]: Can we not say 'heat map'. The visualization we use for tag distributions is not a heat map.

Comment (adhiraj) [reply src/collections/_documentation/enriching-error-data/context.md:21]: 👍

Comment (adhiraj) [reply src/collections/_documentation/enriching-error-data/context.md:21]: @MimiDumpling could we change this to "tag-distribution maps"? (mark if you have a different suggestion let us know)

Comment (MimiDumpling) [reply src/collections/_documentation/enriching-error-data/context.md:21]: Will do. Thanks!

File: src/collections/_documentation/enriching-error-data/context.md [hunk 2]
```diff
@@ -30,61 +50,64 @@ Users consist of a few key pieces of information which are used to construct a u
...
+![Content%20Freeze%20DON%20T%20add%20content%20Docs%20Additional%20D/Screenshot_2020-03-28_19.36.34.png](Content%20Freeze%20DON%20T%20add%20content%20Docs%20Additional%20D/Screenshot_2020-03-28_19.36.34.png)
 
-## Extra Context
+{% include components/platform_content.html content_dir='set-extra' %}
 
-In addition to the structured context that Sentry understands, you can send arbitrary key/value pairs of data which will be stored alongside the event. These are not indexed and are simply used to add additional information about what might be happening:
+In the past, custom context was called "extra" and set via a method like setExtra(), which is deprecated.
 
-{% include components/platform_content.html content_dir='set-extra' %}
+### Context Size Limits (TO REVIEW)
```
Comment (adhiraj) [inline src/collections/_documentation/enriching-error-data/context.md:93]: remove "to review"

File: src/collections/_documentation/enriching-error-data/context.md [hunk 3]
```diff
@@ -30,61 +50,64 @@ Users consist of a few key pieces of information which are used to construct a u
...
+
+Maximum payload size: There are times when you may want to send the whole application state as extra data. Sentry doesn't recommend this as the application state can be extensive and easily exceed the 200kB maximum that Sentry has on individual event payloads. When this happens, you’ll get an `HTTP Error 413 Payload Too Large` message as the server response (when `keepalive: true` is set as `fetch` parameter), or the request will stay in the `pending` state forever (for example, in Chrome).
```
Comment (adhiraj) [inline src/collections/_documentation/enriching-error-data/context.md:95]: @untitaker @bruno-garcia these 2 paragraphs say somewhat different things about what happens to large payloads. do you know how we treat it, and/or how we should rephrase this section?

File: src/collections/_documentation/enriching-error-data/context.md [hunk 4]
```diff
@@ -30,61 +50,64 @@ Users consist of a few key pieces of information which are used to construct a u
...
+{% include components/platform_content.html content_dir='set-tag' %}
 
-You can set the severity of an event to one of five values: `fatal`, `error`, `warning`, `info`, and `debug`. `error` is the default, `fatal` is the most severe, and `debug` is the least.
+Sentry promotes several pieces of predefined data to tags. We strongly recommend you don't set custom tags with these reserved names. But if you do, you can search for them with a [special tag syntax]({%- link _documentation/workflow/search.md -%}).
 
-{% include components/platform_content.html content_dir='set-level' %}
+### Context
 
-## Setting the Fingerprint
+Custom contexts allow you to attach arbitrary data (strings, lists, dictionaries) to an event. These are unsearchable, but are viewable in the "Additional Data" section of an event.
```
Comment (adhiraj) [inline src/collections/_documentation/enriching-error-data/context.md:85]: "...These are unsearchable, but are viewable **on the issue page.**"

File: src/collections/_documentation/enriching-error-data/context.md [hunk 5]
```diff
@@ -30,61 +50,64 @@ Users consist of a few key pieces of information which are used to construct a u
...
 
-Sentry uses one or more "fingerprints" to decide how to group errors into issues. More information can be found [here]({%- link _documentation/data-management/event-grouping/index.md -%}).
+![Content%20Freeze%20DON%20T%20add%20content%20Docs%20Additional%20D/Screenshot_2020-03-28_19.36.34.png](Content%20Freeze%20DON%20T%20add%20content%20Docs%20Additional%20D/Screenshot_2020-03-28_19.36.34.png)
 
-## Extra Context
+{% include components/platform_content.html content_dir='set-extra' %}
 
-In addition to the structured context that Sentry understands, you can send arbitrary key/value pairs of data which will be stored alongside the event. These are not indexed and are simply used to add additional information about what might be happening:
+In the past, custom context was called "extra" and set via a method like setExtra(), which is deprecated.
```
Comment (adhiraj) [inline src/collections/_documentation/enriching-error-data/context.md:91]: Let's add:

Custom data set using `setExtra` appears in the "Additional Data" section of an event, whereas each key set using `setContext` gets its own section on the issue page, with the section title being the key name.

File: src/collections/_documentation/enriching-error-data/context.md [hunk 6]
```diff
@@ -30,61 +50,64 @@ Users consist of a few key pieces of information which are used to construct a u
...
```
Comment (adhiraj) [inline src/collections/_documentation/enriching-error-data/context.md:92]: Before the Context size limits section let's add:

For more detail on predefined and custom contexts, see https://docs.sentry.io/development/sdk-dev/event-payloads/contexts/.

**Comment by MimiDumpling** [comment]
Hello everyone! I'm having trouble with the tests passing. I suspect it's the way I handled the `include` changes, but I'm not sure what I did wrong. Will someone help?

Both my snapshot test and my docker test failed. Pretty sure it's related to Travis failing.

**Comment by MimiDumpling** [comment]
@HazAT will you take a look at why the tests are failing? I suspect it's because I did the `include` statements wrong. I renamed the page (context.md -> additional-data.md) and perhaps the `include` statements are dependent on the page's name?

**Comment by MimiDumpling** [comment]
> Made a pass based on feedback
> 
> @MimiDumpling could we also backreference this new page from https://docs.sentry.io/development/sdk-dev/event-payloads/contexts/. I'm imaging a line at the end of the first paragraph saying "See [Additional Data] for an overview of sending additional data with your event."

Done! 😸

**Comment by MimiDumpling** [comment]
@adhiraj The tests aren't passing and I'm not sure why. Will you help or find the appropriate person to help?

**Comment by MimiDumpling** [comment]
> @HazAT will you take a look at why the tests are failing? I suspect it's because I did the `include` statements wrong. I renamed the page (context.md -> additional-data.md) and perhaps the `include` statements are dependent on the page's name?

@adhiraj ^ that's my hypothesis for what went wrong.

File: src/collections/_documentation/enriching-error-data/additional-data.md [hunk 7]
```diff
@@ -0,0 +1,117 @@
...
+example_tag_name: page_locale
+example_tag_value: de-at
+example_user_email: 'john.doe@example.com'
+example_extra_key: character_name
+example_extra_value: 'Mighty Fighter'
+---
+
+Additional data attached to an event can be predefined or custom.
+
+[Predefined data](#predefined-data) is data that Sentry recognizes and uses to enhance the user experience. For example, Sentry's concept of a user lets us show how many unique users are affected by an issue. Sentry automatically captures predefined data where possible, such as information about the browser, HTTP request, and OS in the browser Javascript SDK.
```
Comment (joshuarli) [inline src/collections/_documentation/enriching-error-data/additional-data.md:13]: @MimiDumpling JavaScript, not Javascript. That's what the test is yelling about.

Comment (MimiDumpling) [reply src/collections/_documentation/enriching-error-data/additional-data.md:13]: Awesome! Thanks for that catch. I'll fix it shortly.

**Pull request merged**